src/Controller/GenieController.php

```diff
@@ -144,11 +144,12 @@ public function testInput(array $inputs): array
         return $errors;
     }
 
-    public function showAllGenies()
+    public function showAllGenies($parameter)
```

Comment:
Bad naming => you call it speciality in routes so you should call it speciality in the function